File: test/coop/rchain/lang/lexer/RholangLexTest.java
package coop.rchain.lang.lexer;

import com.intellij.lexer.Lexer;
import com.intellij.openapi.application.PathManager;
import com.intellij.testFramework.LexerTestCase;
import coop.rchain.RholangTestUtil;
import coop.rchain.lang.RhoLexerAdapter;

import java.nio.file.Paths;

public class RholangLexTest extends LexerTestCase {

  @Override
  protected Lexer createLexer() {
    return new RhoLexerAdapter();
  }

  @Override
  protected String getDirPath() {
    return Paths.get(RholangTestUtil.getBaseTestDataPath(), "lexer").toString();
  }

  public void testToken() {
    doFileTest("rho");
  }
}

File: test/coop/rchain/lang/lexer/RholangLexTest.kt
//package coop.rchain.lang.lexer
//
//import com.intellij.lexer.Lexer
//import com.intellij.testFramework.LexerTestCase
//import coop.rchain.RholangTestUtil
//import coop.rchain.lang.RhoLexerAdapter
//
//import java.nio.file.Paths
//
//class RholangLexTest : LexerTestCase() {
//
//  override fun createLexer(): Lexer {
//    return RhoLexerAdapter()
//  }
//
//  override fun getDirPath(): String =
//    Paths.get(RholangTestUtil.baseTestDataPath, "lexer").toString()
//
//  override fun getPathToTestDataFile(extension: String): String =
//    this.getDirPath() + "/" + this.getTestName(true) + extension
//
//  fun testToken() {
//    doFileTest("rho")
//  }
//}
